src/modules/hg_branch.rs

```diff
@@ -74,26 +73,36 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     Some(module)
 }
 
-fn get_hg_branch_name(ctx: &Context) -> String {
-    std::fs::read_to_string(ctx.current_dir.join(".hg").join("branch"))
-        .map_or_else(|_| "default".to_string(), |s| s.trim().into())
+fn get_hg_repo_root<'a>(ctx: &'a Context) -> Result<&'a Path, Error> {
+    let dir = ctx.current_dir.as_path();
+    let dev_id = dir.device_id();
+    for root_dir in dir.ancestors() {
+        if dev_id != root_dir.device_id() {
+            break;
+        }
+        if root_dir
+            .read_dir()?
+            .any(|e| e.unwrap().file_name() == ".hg")
+        {
```

Comment:
Please avoid `unwrap`, please just use this:
```suggestion
        if root_dir.join(".hg").is_dir() {
```

Reply:
BTW why should unwrap be avoided? I'm new to rust and I'd like to know why unwrap should be avoided (just like the other change in `device_id`).